packages/site/cms-site/src/iframebridge/Preview.tsx

```diff
@@ -1,28 +1,49 @@
 import * as React from "react";
 import scrollIntoView from "scroll-into-view-if-needed";
+import styled from "styled-components";
 
-import { ComponentTitle, Root, Selection } from "./Preview.sc";
 import { useIFrameBridge } from "./useIFrameBridge";
+import { BLOCK_PREVIEW_CONTAINER_DATA_ATTRIBUTE } from "./utils";
 
-interface PreviewProps {
+interface Props {
     adminRoute: string;
-    type: string;
+    label: string;
     enabledAutoScrolling?: boolean;
 }
 
-export const Preview: React.FunctionComponent<PreviewProps> = ({ adminRoute, type, children, enabledAutoScrolling = true }) => {
+export const Preview: React.FunctionComponent<Props> = ({ adminRoute, children, label, enabledAutoScrolling = true }) => {
     const iFrameBridge = useIFrameBridge();
     const isSelected = adminRoute === iFrameBridge.selectedAdminRoute;
     const isHovered = adminRoute === iFrameBridge.hoveredAdminRoute;
-    const rootEl = React.useRef<HTMLDivElement | null>(null);
+    const previewElementContainerRef = React.useRef<HTMLDivElement>(null);
+
+    React.useEffect(() => {
+        const previewElement = previewElementContainerRef.current
+            ? {
+                  label,
+                  adminRoute,
+                  element: previewElementContainerRef.current,
+              }
+            : null;
+
+        if (previewElement) {
+            iFrameBridge.addPreviewElement(previewElement);
+        }
+
+        return () => {
+            if (previewElement) {
+                iFrameBridge.removePreviewElement(previewElement);
+            }
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is it safe to disable the rule here? Could you add a comment explaining why?

Reply:
Probably but I refactored the code to make this not necessary.